Jenkinsfile-bleeding-edge-monthly

```diff
@@ -10,42 +15,102 @@ def runTests(String testPath, boolean jumbo = false) {
+            echo LDFLAGS+="-L/usr/local/lib -Wl,-R/usr/local/lib -I/usr/local/include/c++/v1  -L/usr/local/lib -Wl,-rpath,/usr/local/lib -lc++ -lc++abi -lm -lc -lgcc_s -lgcc ${ldFlags}" >> ${makeLocalPath}
+        """
+    }
+    sh "${cxx} --version"
+}
+
+def runPerformanceTests(String testsPath, String stancFlags = "", String compiler = "clang"){
+    unstash 'ubuntu-exe'
+
+    sh """
+        git clone --recursive --depth 50 https://github.com/stan-dev/performance-tests-cmdstan
+    """
+
+    utils.checkout_pr("cmdstan", "performance-tests-cmdstan/cmdstan", params.cmdstan_pr)
+    utils.checkout_pr("stan", "performance-tests-cmdstan/cmdstan/stan", params.stan_pr)
+    utils.checkout_pr("math", "performance-tests-cmdstan/cmdstan/stan/lib/stan_math", params.math_pr)
+
+    if (compiler == "gcc"){
+        setupMakeLocal("0","gcc","g++","","","", "-ltbb", "performance-tests-cmdstan/cmdstan/make/local")
+    }
+    else if (compiler == "clang"){
+        setupMakeLocal("0","clang","clang++","","","", "-ltbb", "performance-tests-cmdstan/cmdstan/make/local")
+    }
+
+    sh """
+        cd performance-tests-cmdstan
+        mkdir cmdstan/bin
+        cp ../bin/stanc cmdstan/bin/linux-stanc
+        cd cmdstan; make clean-all;
+    """
+
+    if (stancFlags?.trim()) {
+        sh "cd performance-tests-cmdstan/cmdstan && echo 'STANCFLAGS= $stancFlags' >> make/local"
+    }
+
+    sh """
+        cd performance-tests-cmdstan/cmdstan
+        echo 'O=0' >> make/local
+        make -j${env.PARALLEL} build; cd ..
+        python3 runPerformanceTests.py -j${env.PARALLEL} --runs=0 ${testsPath}
+    """
+}
 
 pipeline {
     agent none
     options {
         skipDefaultCheckout()
         preserveStashes(buildCount: 7)
-        parallelsAlwaysFailFast()
     }
     parameters {
-        booleanParam(defaultValue: false, name: 'disableJumbo', description: 'Disable Jumbo tests. This takes longer and should only be used for debugging if it is believed that the jumbo tests are causing failures.')
+        booleanParam(defaultValue: false, name: 'buildDocker', description: 'Build docker image with latest gcc, clang')
+        string(defaultValue: 'nightly', name: 'stanc3_bin_url', description: 'Custom stanc3 binary url')
+        string(defaultValue: 'develop', name: 'cmdstan_pr',
+               description: "CmdStan PR to test against. Will check out this PR in the downstream Stan repo.")
+        string(defaultValue: 'develop', name: 'stan_pr',
+               description: "Stan PR to test against. Will check out this PR in the downstream Stan repo.")
+        string(defaultValue: 'develop', name: 'math_pr',
+               description: "Math PR to test against. Will check out this PR in the downstream Math repo.")
+        string(defaultValue: 'master', name: 'stanc3_pr',
+               description: "Stanc3 PR to test against. Will check out this PR in the downstream Math repo.")

```

Comment:
Why do we have both a `stanc3_bin_url` parameter and a `stanc3_pr` parameter. Are both necessary?